open_spiel/games/gin_rummy.h

```diff
@@ -35,53 +35,71 @@
 //  "knock_card"      int    set a specific knock card     (default = 10)
 //  "gin_bonus"       int    bonus for getting gin         (default = 25)
 //  "undercut_bonus"  int    bonus for an undercut         (default = 25)
+//  "num_ranks"       int    num ranks in deck             (default = 13)
+//  "num_suits"       int    num suits in deck             (default = 4)
+//  "hand_size"       int    num cards in player hand      (default = 10)
 
 #include <memory>
 #include <string>
 #include <utility>
 #include <vector>
 
 #include "open_spiel/abseil-cpp/absl/types/optional.h"
+#include "open_spiel/fog/observation_history.h"
 #include "open_spiel/games/gin_rummy/gin_rummy_utils.h"
+#include "open_spiel/observer.h"
 #include "open_spiel/spiel.h"
 
 namespace open_spiel {
 namespace gin_rummy {
 
+// Game constants
+inline constexpr int kDefaultNumRanks = 13;
+inline constexpr int kDefaultNumSuits = 4;
+inline constexpr int kDefaultNumCards = 52;
 inline constexpr int kNumPlayers = 2;
 inline constexpr int kMaxPossibleDeadwood = 98;  // E.g. KsKcQdQhJsJcTdTh9s9c
 inline constexpr int kMaxNumDrawUpcardActions = 50;
-inline constexpr int kHandSize = 10;
+inline constexpr int kDefaultHandSize = 10;
+// TODO(jhtschultz) should kMaxStockSize be a tight upper bound for nonstandard
+// deck size?
 inline constexpr int kMaxStockSize = 31;  // Stock size when play begins
 inline constexpr int kWallStockSize = 2;
 inline constexpr int kDefaultKnockCard = 10;
 inline constexpr int kDefaultGinBonus = 25;
 inline constexpr int kDefaultUndercutBonus = 25;
+// Action constants
 inline constexpr int kDrawUpcardAction = 52;
 inline constexpr int kDrawStockAction = 53;
 inline constexpr int kPassAction = 54;
 inline constexpr int kKnockAction = 55;
 inline constexpr int kMeldActionBase = 56;  // First lay meld action
 inline constexpr int kNumMeldActions = 185;
 inline constexpr int kNumDistinctActions = kMeldActionBase + kNumMeldActions;
+// Observer constants
 inline constexpr int kObservationTensorSize =
-    kNumPlayers          // Player turn
-    + kDefaultKnockCard  // Knock card
-    + kNumCards          // Player hand
-    + kNumCards          // Upcard
-    + kNumCards          // Discard pile
-    + kMaxStockSize      // Stock size
-    + kNumMeldActions;   // Opponent's layed melds
+    kNumPlayers             // Player turn
+    + kDefaultKnockCard     // Knock card
```

Comment:
Oops I thought I replied to this earlier! This is here because the observation tensor size is fixed and I had seen other games define a fixed tensor size here (backgammon). We could just as easily run this calculation inside of ObservationTensorShape(). Your call.